bookstore/clone.py

```diff
@@ -0,0 +1,112 @@
+import json
+import os
+
+import aiobotocore
+
+from notebook.base.handlers import IPythonHandler
+from tornado import web
+from jinja2 import FileSystemLoader
+
+from . import PACKAGE_DIR
+from .s3_paths import s3_path
+from .bookstore_config import BookstoreSettings
+
+
+BOOKSTORE_FILE_LOADER = FileSystemLoader(PACKAGE_DIR)
+
+
+class BookstoreCloneHandler(IPythonHandler):
+    """Clone a notebook from s3."""
+
+    def initialize(self):
+        # create an easy helper to get at our bookstore settings quickly
+        self.bookstore_settings = BookstoreSettings(config=self.config)
+
+        self.session = aiobotocore.get_session()
+
+    async def _clone(self, s3_bucket, file_key):
+        path = file_key
+
+        self.log.info(f"bucket: {s3_bucket}")
+        self.log.info(f"key: {file_key}")
+        full_s3_path = s3_path(s3_bucket, path)
+
+        async with self.session.create_client(
+            's3',
+            aws_secret_access_key=self.bookstore_settings.s3_secret_access_key,
```

Comment:
Do we force the config to be loaded into `bookstore_settings`? There's many instances where these credentials are fetch JIT from the system config loader. Likely a conversation out of scope of this PR though.